fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/consumers/TranslationConsumer.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1.provider.consumers;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import net.minecraft.block.Block;
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.attribute.EntityAttribute;
+import net.minecraft.entity.effect.StatusEffect;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.stat.StatType;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.datagen.v1.provider.FabricLanguageProvider;
+
+/**
+ * A consumer used by {@link FabricLanguageProvider#generateTranslations(TranslationConsumer)}.
+ */
+@FunctionalInterface
+public interface TranslationConsumer {
```

Comment:
Looking good, are we sure TranslationConsumer is the right name for this? Yarn names the similar vanilla classes `BlockStateModelGenerator` and `ItemModelGenerator`

The provider you added is also uses `Language` and not `Translation`

Reply:
Maybe consumer is correct then its not the end of the world? It doesnt need the `@FunctionalInterface` annotation tho. Could also work just fine being an inner class, no need to add it to its own package.

Reply:
It is functional my bad.

Reply:
Ok stick with it then.👍 I think it might be best as an inner class but I’ll leave that up to you.